business/istio_validations.go

```diff
@@ -63,23 +63,20 @@ func (in *IstioValidationsService) GetValidations(ctx context.Context, namespace
 	errChan := make(chan error, 1)
 
 	var istioConfigList models.IstioConfigList
-	var exportedResources kubernetes.ExportedResources
 	var services models.ServiceList
 	var namespaces models.Namespaces
 	var workloadsPerNamespace map[string]models.WorkloadList
 	var mtlsDetails kubernetes.MTLSDetails
 	var rbacDetails kubernetes.RBACDetails
 	var registryServices []*kubernetes.RegistryService
 
-	wg.Add(8) // We need to add these here to make sure we don't execute wg.Wait() before scheduler has started goroutines
+	wg.Add(6) // We need to add these here to make sure we don't execute wg.Wait() before scheduler has started goroutines
 
 	// We fetch without target service as some validations will require full-namespace details
-	go in.fetchIstioConfigList(ctx, &istioConfigList, namespace, errChan, &wg)
-	go in.fetchExportedResources(ctx, &exportedResources, namespace, errChan, &wg)
+	go in.fetchIstioConfigList(ctx, &istioConfigList, &mtlsDetails, &rbacDetails, namespace, errChan, &wg)
 	go in.fetchNamespaces(ctx, &namespaces, errChan, &wg)
 	go in.fetchAllWorkloads(ctx, &workloadsPerNamespace, errChan, &wg)
-	go in.fetchNonLocalmTLSConfigs(ctx, &mtlsDetails, namespace, errChan, &wg)
-	go in.fetchAuthorizationDetails(ctx, &rbacDetails, namespace, errChan, &wg)
+	go in.fetchNonLocalmTLSConfigs(&mtlsDetails, errChan, &wg)
 	go in.fetchServices(ctx, &services, namespace, errChan, &wg)
```

Comment:
I'm reviewing the "GetValidations" method and probably we don't need to keep using the "fetchServices" at this level.
"GetValidations" is used for "IstioConfigList", "NamespaceValidationSummary", "ServiceDetails" and "ServiceUpdate".

Only in these two cases, the "service" parameter is used then no need to use the "fetchServices" unless it's under that condition.

That would potentially improve the other cases that are more common.